File: config/config-sbc2ha-new/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/action/MqttActionConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.action;

import com.dfi.sbc2ha.config.sbc2ha.definition.enums.ActionType;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public class MqttActionConfig extends ActionConfig {

    String topic;

    @JsonProperty("action_mqtt_msg")
    String actionMqttMsg;

    public MqttActionConfig() {
        super();
        action = ActionType.MQTT;
    }
}

File: config/editor/src/app/definition/action/MqttActionConfig.ts
import {ActionConfig} from "./ActionConfig";
import {ActionType} from "../enums/ActionType";
import {Expose} from "class-transformer";


export class MqttActionConfig extends ActionConfig {
    topic: string;
    @Expose({name: "action_mqtt_msg"})
    actionMqttMsg: string;

    constructor(topic: string, actionMqttMsg: string) {
        super(ActionType.MQTT);
        this.topic = topic;
        this.actionMqttMsg = actionMqttMsg;
    }
}
